### PR TITLE
fix: Only delete file when it exists

### PR DESCRIPTION
**Pull request by AlexV525**
### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dartlang.org/packages/dio)
- [x] I have searched for a similar pull request in the [project](https://github.com/flutterchina/dio/pulls) and found none
- [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
- [x] I have added the required tests to prove the fix/feature I am adding
- [x] I have updated the documentation (if necessary)
- [x] I have run the tests and they pass

This merge request fixes / refers to the following issues: N/A

### Pull Request Description

Before we delete the downloading file, we should determine whether it exists.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
Nice, thanks! Can you fix the formatting and add a changelog entry please.

**Comment by AlexV525** [comment]
Should be good to go.

File: dio/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 - require Dart `2.12.1` which fixes exception handling for secure socket connections (https://github.com/dart-lang/sdk/issues/45214)  
 - `ResponseBody.statusCode` is now non-nullable
 
+# 4.0.5-beta2
```
Comment (kuhnroyal) [inline dio/CHANGELOG.md:5]:
```suggestion
```

**Review by kuhnroyal — CHANGES_REQUESTED** [review]
We don't know yet what the next version will be. Just add the changes to the unreleased block.

**Review by kuhnroyal — APPROVED** [review]

**Comment by AlexV525** [comment]
Done with rebasing, squashing, and format update.

**Comment by kuhnroyal** [comment]
Thank you!

**Pull request merged**